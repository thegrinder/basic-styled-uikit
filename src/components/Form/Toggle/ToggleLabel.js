import styled from 'styled-components';

const ToggleLabel = styled.label`
  display: inline-block;
  width: 100%;
`;

export default ToggleLabel;
