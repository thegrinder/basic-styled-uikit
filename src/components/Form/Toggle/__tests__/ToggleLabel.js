import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import ToggleLabel from '../ToggleLabel';

const children = <span>children</span>;
const renderComponent = () => render(
  <ToggleLabel>{children}</ToggleLabel>,
);

describe('<ToggleLabel />', () => {
  afterEach(cleanup);

  it('should render correctly with children', () => {
    const { container: { firstChild }, getByText } = renderComponent();
    const childrenElement = getByText('children');
    expect(firstChild).toBeDefined();
    expect(firstChild).toContainElement(childrenElement);
    expect(firstChild).toMatchSnapshot();
  });

  it('should render <label> tag', () => {
    const { container: { firstChild } } = renderComponent();
    expect(firstChild.tagName).toEqual('LABEL');
  });
});
