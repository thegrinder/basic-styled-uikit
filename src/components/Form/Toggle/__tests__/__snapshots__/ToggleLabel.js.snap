// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleLabel /> should render correctly with children 1`] = `
<label
  class="ToggleLabel-sc-1p6qi5g-0 eRLkxH"
>
  <span>
    children
  </span>
</label>
`;
